### ISSUE TITLE
Link Catching Not Working

### ISSUE THREAD

**Issue by gitman0** [bogenpirat/remote-torrent-adder#189]
It seems like since the Dec 20 update, the link catcher no longer works on my favorite torrent site. What information can I gather to troubleshoot this issue?

Thanks

**Comment by bogenpirat** [comment]
let's start off with the basics. what sort of malfunction you're experiencing: what are you doing, step by step, and what you're expecting to happen that isn't.

**Comment by gitman0** [comment]
when I left click a torrent link, I get the "save as" dialog, where previously the addon would forward it over to my transmission client. the addon icon is also no longer visible in the omnibus when on the page that has torrent download links (where it was previously visible). I can right click the torrent link and send it to the transmission client - that still works - just not the automatic process of left clicking the link / forwarding to transmission client.

**Comment by gitman0** [comment]
should say omnibox, sorry

**Comment by bogenpirat** [comment]
icons don't get displayed inside the omnibar anymore since over 10 versions back, they're all next to the omnibar these days. if it doesn't display for you at all (normal states are greyed out or green if a link was found), it might be hidden in the three dots menu.

the fact that chrome downloads the torrent as opposed to adding it to the torrent client might be because of filters. i don't see anything in the pre-december 20th commits that change anything about how that works, so maybe you should paste a link to a torrent so we can figure out whether your torrent site's links aren't caught by the filters anymore. oh, and also post your filters from the link catching tab please.

**Comment by gitman0** [comment]
i can resize the omnibar to make it hide the icons that are being displayed - it is definitely showing all of them currently, sans the torrent-adder icon i was used to seeing... and just like you mentioned, it used to be green when torrents are found, grey when nothing.

Filters - I added the last one myself, and I did not pay close attention to whether the first 2 that were already there would match the link format:

http://gitman.me/images/rta-filters.png

Example link (with surrounding markup):

`<div><a href="download.php/2143834/Silicon.Cowboys.2016.LIMITED.DVDRip.x264-BiPOLAR.torrent"><i class="fa fa-download dlBtn" title="Download" alt="Download"></i></a></div>`

On the GET response, the site responds with the following content-disposition header - take note that the file name has changed slightly, I'm not sure if this matters:

`content-disposition:attachment; filename="Silicon.Cowboys.2016.LIMITED.DVDRip.x264-BiPOLAR [ABC].torrent"`

Thanks for your help thus far

**Comment by bogenpirat** [comment]
then we're looking at two separate issues. i'd guess you've accidentally hidden the icon? idk, i've never had that issue and i don't think it's a problem with the extension since the manifest/options setting already force the existence of the button, lest you fuck around with that.

as for the primary issue: filter looks good (unless you've hidden like a space or something at the end of that last one), so really the interesting bit would be figuring out whether the extension registers any links from that page.

the result of the GET doesn't matter too much since all of that is delegated to the XHR anyway, and any resulting headers don't really matter at all. for illustration, RTA will try adding a .jpg with `Content-Type: application/frenchfries` if your filters match it and an XMLHttpRequest can download it off the server.

so i'm guessing you should try taking a stab at finding the RTA icon again in order to verify that RTA does catch links. check the options page (`chrome-extension://oabphaconndgibllomdcjbfdghcmenci/options.html`), there's an item called "Address bar indicator" that should be enabled. if that doesn't work, just remove and re-add the extension.

**Comment by gitman0** [comment]
if the icon is hidden, chrome is not giving me the option to unhide it in the extensions menu.

i went through removing the extension and re-adding it. after it was installed, the icon appeared in the omnibar and showed a box that said the icon would appear there whenever the page supports it (paraphrased). i am running chrome 38 on this machine, so your comment about that changing 10 versions back i am now realizing is probably in reference to the chrome version, not the addon version.

i tested it out in incognito mode as well, still no dice.

is there a way to debug what the addon itself is doing?

**Comment by gitman0** [comment]
Hello,

Today I noticed an error showing up in the javascript console:

_Uncaught ReferenceError: Invalid left-hand side in assignment_

And it referenced line 42 of content_rta.js, which is

`[rta_modal_open, rta_modal_close] = rta_modal_init();`

Destructured assignment was not available until Chrome 49 (https://www.chromestatus.com/feature/4588790303686656), so I replaced that line (and line 27) with this:

```
init = rta_modal_init();
rta_modal_open = init[0];
rta_modal_close = init[1];
```

Now I get the icon in the omnibar, and the link catcher is working fine.

**Comment by bogenpirat** [comment]
ah, that makes sense for the original issue. but why would you cling to a 1 year old chrome version?

**Comment by gitman0** [comment]
a slew of reasons, mainly being that this is a personal laptop that i also use for work, running a older 32-bit fedora install (chrome 49+ is 64-bit only), and i can't afford the downtime to update it.

**Comment by bogenpirat** [comment]
gotcha. glad you figured it out. i'm matching the fix in 67f1840, thanks for the sleuthing. release isn't a priority right now.
